crates/ruff_linter/src/rules/flake8_simplify/rules/if_else_block_instead_of_if_exp.rs

```diff
@@ -121,15 +132,19 @@ pub(crate) fn if_else_block_instead_of_if_exp(checker: &mut Checker, stmt_if: &a
         return;
     }
 
-    let target_var = &body_target;
-    let ternary = ternary(target_var, body_value, test, else_value);
-    let contents = checker.generator().stmt(&ternary);
+    let possible_replacement = format!(
+        "{} = {} if {} else {}",
+        locator.slice(body_target),
+        locator.slice(&**body_value),
```

Comment:
I tend to do `.as_ref()` here instead, but maybe @BurntSushi does not like that.

Reply:
I'd say the code as written here is idiomatic. Basically, what you have here is a `&Box<Expr>` and you want a `&Expr`. While it's true that `as_ref()` will work here, it only works because the set of possible trait impls that apply is exactly `1`. If the number of trait impls increases such that it becomes ambiguous, the code will stop compiling. I think in _this_ case, it _seems_ like the relevant trait impls are very unlikely to increase. Which makes `as_ref()` in this particular case probably pretty safe. The problem is that adjudicating which `as_ref()` calls are probably fine and will never break, and which are not is IMO extremely tricky. So it's better to just avoid it unless you have an explicit `AsRef<TargetType>` in scope somewhere (or some other explicit annotation that makes the `as_ref()` target unambiguous).